### PR TITLE
[ refactor ] Rename Data.Strings to Data.String (1/2)

### PR DESCRIPTION
**Pull request by andylokandy**
The reason why I submit this PR is that the other modules in `Data` are singular and the module name `Data.Strings` is out of sync with `Data.String.Extra` in `contrib`. Therefore, I think this will make the naming more consistent.

But this is indeed a big breaking change, we may break many many existing Idris1 and Idris2 codes... So I think it has to be thought carefully before being accepted.

### CODE REVIEW

**Comment by gallais** [comment]
This makes it impossible to build from the previous version so won't be
accepted as is. A move towards the end result of this PR is to first
move `Data.Strings` to `Data.String` and then reexport publicly the content
in `Data.Strings`.

After a minor release has happened, you'll be able to rename the `Data.Strings`
import to `Data.String` ones without breaking the build chain.

File: libs/base/Data/Strings.idr [hunk 1]
```diff
@@ -1,3 +1,4 @@
+||| To be deprecated
 module Data.Strings
```
Comment (gallais) [inline libs/base/Data/Strings.idr:3]: Can we not replace the content of this module with `import public Data.String`?

Comment (andylokandy) [reply libs/base/Data/Strings.idr:3]: This was problematic on my machine. But I seems not able to reproduce it... I'll push it here and let the CI try it.

Comment (andylokandy) [reply libs/base/Data/Strings.idr:3]:
```
Error: While processing right hand side of show. Undefined name Strings.singleton. 

System/Path.idr:110:22--110:44
     |
 110 |   show (Disk disk) = Strings.singleton disk ++ ":"
     |                      ^^^^^^^^^^^^^^^^^^^^^^
```

Comment (andylokandy) [reply libs/base/Data/Strings.idr:3]:
```
Error: While processing right hand side of show. Undefined name Strings.singleton. 

System/Path.idr:110:22--110:44
     |
 110 |   show (Disk disk) = Strings.singleton disk ++ ":"
     |                      ^^^^^^^^^^^^^^^^^^^^^^
```

Comment (gallais) [reply libs/base/Data/Strings.idr:2]: The trick was to import qualified! ;)

**Pull request merged**